zerver/actions/users.py

```diff
@@ -516,6 +745,12 @@ def do_deactivate_user(
                 RealmAuditLog.ROLE_COUNT: realm_user_count_by_role(user_profile.realm),
             },
         )
+
+        if group_setting_updates is not None:
```

Comment:
We would also need to remove deactivated bots here from the anonymous groups. So, would need to modify the `check_group_permission_updates_for_deactivating_user` function to handle multiple user_profile.